pkg/controller/apiextensions/claim/configurator.go

```diff
@@ -43,14 +43,28 @@ const errUnsupportedClaimSpec = "composite resource claim spec was not an object
 // derived from the supplied claim, as {name}-{random-string}. The claim's
 // external name annotation, if any, is propagated to the composite resource.
 func Configure(_ context.Context, cm resource.CompositeClaim, cp resource.Composite) error {
-	cp.SetGenerateName(fmt.Sprintf("%s-", cm.GetName()))
+	// It's possible we're being asked to configure a statically provisioned XR,
+	// in which case we should respect its existing name and external name.
+	en := meta.GetExternalName(cp)
+	if cp.GetName() == "" {
+		cp.SetGenerateName(fmt.Sprintf("%s-", cm.GetName()))
+	}
+
 	meta.AddAnnotations(cp, cm.GetAnnotations())
 	meta.AddLabels(cp, cm.GetLabels())
 	meta.AddLabels(cp, map[string]string{
 		xcrd.LabelKeyClaimName:      cm.GetName(),
 		xcrd.LabelKeyClaimNamespace: cm.GetNamespace(),
 	})
 
+	// If our XR is already named we assume it exists and was statically
+	// provisioned, in which case we want to restore its original external name
+	// (even if that external name was empty) in order to ensure we don't try to
+	// rename anything after the fact.
+	if cp.GetName() != "" {
```

Comment:
Right after I opened this it occurred to me we could also use `meta.WasCreated` as a more accurate sign of whether the XR exists. It's probably functionally about the same.